great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -2185,9 +2188,14 @@ def save_profiler(
         name: Optional[str] = None,
         ge_cloud_id: Optional[str] = None,
     ) -> None:
-        key: Union[
-            GeCloudIdentifier, ConfigurationIdentifier
-        ] = self.profiler_store.determine_key(name=name, ge_cloud_id=ge_cloud_id)
+        key: Union[GeCloudIdentifier, ConfigurationIdentifier]
+        if self.ge_cloud_mode:
+            key = GeCloudIdentifier(
+                resource_type=GeCloudRESTResource.PROFILER, ge_cloud_id=ge_cloud_id
+            )
+        else:
+            key = ConfigurationIdentifier(configuration_key=name)
```

Comment:
Both of these ideas below might be bad. But even if they are good ideas, I don't think they should stop merging 😄 .

1. How hard would it be to use some polymorphism to derive the key rather than all these if/else blocks?

Like move the id determining logic to the `RuleBasedProfiler.identifier` property.

```python
    def save_profiler(
        self,
        profiler: RuleBasedProfiler,
        name: Optional[str] = None,
        ge_cloud_id: Optional[str] = None,
    ) -> None:
        self.profiler_store.set(key=profile.key, value=profiler.config)
```

2. Update `ConfigurationIdentifier.__init__` (or `__new__`) so that it can generate `GeCloudIdentifier` as well. Which prevents us from having so much conditional `if cloud else:` logic.

Reply:
oh I definitely agree I don't love the changes here 😬 

I wanted to do the minimal required changes just to get it working but this overall feature needs some TLC. Making modifications to the actual profiler object may cause some issues so I opted for the less elegant solution here.

I believe we're eventually going to add an id onto the profiler itself which should be able to remedy most of this. Would love to hear what other people think though maybe now is the time for such changes?

Reply:
but this will unblock Cloud for the time being